xoa/data_source_vms.go

```diff
@@ -0,0 +1,107 @@
+package xoa
+
+import (
+	"log"
+	"strings"
+
+	"github.com/ddelnano/terraform-provider-xenorchestra/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceXoaVms() *schema.Resource {
+
+	return &schema.Resource{
+		Read: dataSourceVmsRead,
+		Schema: map[string]*schema.Schema{
+			"vms": &schema.Schema{
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     resourceVm(),
+			},
+			"num": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"pool_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"container": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"power_state": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func dataSourceVmsRead(d *schema.ResourceData, m interface{}) error {
+	c := m.(client.XOClient)
+	searchVm := client.Vm{
+		PowerState: d.Get("power_state").(string),
+		Container:  d.Get("container").(string),
+		PoolId:     d.Get("pool_id").(string),
+	}
+
+	vms, err := c.GetVms(searchVm)
+	if err != nil {
+		return err
+	}
+
+	if err = d.Set("vms", vmToMapList(vms)); err != nil {
+		return err
+	}
+	if err = d.Set("num", len(vms)); err != nil {
+		return err
+	}
```

Comment:
I think we should encourage people to use the [length function](https://www.terraform.io/docs/language/functions/length.html) instead (`length(data.xenorchestra_vms.vms)`). Is there a reason that couldn't satisfy the use cases you had in mind?